views/js/scoring/processor/expressions/preprocessor.js

```diff
@@ -33,26 +33,32 @@ define([
      */
     var preProcessor = {
 
-        /**
-         * Take the operands, cast the values to integer or float, flatten them is collection are given and filter on unwanted values.
-         * @param {Array<ProcessingValue>|Arrat<Array<ProcessingValue>>} operands - to map
-         * @returns {Object} the LODASH wrapper in order to chain on it.
-         */
-        mapNumbers : function mapNumbers(operands){
+        
+        castTypes: function (operands) {
```

Comment:
Jsdoc comments are also missing.
Try also to name your functions:
`castTypes: function castTypes (operands)`